Assets/MRTK/SDK/Experimental/InteractiveElement+StateVisualizer/StateVisualizer/StateAnimatableProperties/AnimatableProperties/ScaleOffsetStateAnimatableProperty.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Interaction
+{
+    /// <summary>
+    /// The ScaleOffset animatable property adds/sets keyframes for the "localScale" property in an animation clip.
+    /// </summary>
+    public class ScaleOffsetStateAnimatableProperty : StateAnimatableProperty
+    {
+        [SerializeField]
+        [Tooltip("The scale offset added to the current scale of the target object.")]
+        private Vector3 scaleOffset;
+
+        /// <summary>
+        /// The scale offset added to the current scale of the target object.
+        /// </summary>
+        public Vector3 ScaleOffset
+        {
+            get => scaleOffset;
+            set => scaleOffset = value;
+        }
+
+        public ScaleOffsetStateAnimatableProperty()
+        {
+            AnimatablePropertyName = "ScaleOffset";
+        }
+
+        public override void SetKeyFrames(AnimationClip animationClip)
+        {
+            if (Target != null)
+            {
+                string targetPath = GetTargetPath(Target);
+
+                AnimationCurve curveX = AnimationCurve.EaseInOut(0, Target.transform.localScale.x, AnimationDuration, Target.transform.localScale.x + ScaleOffset.x);
```

Comment:
I'm seeing a lot of repeated code in all these AnimatableProperty files. Not a huge deal, but it occured to me that you could probably refactor this to avoid repeating yourself, something like utilizing a base class, or some sort of utility method.